code/modules/projectiles/guns/smgs.dm

```diff
@@ -687,3 +687,10 @@
 	if(.)
 		click_empty(user)
 	return FALSE
+
+/obj/item/weapon/gun/smg/nailgun/compact/tactical
```

Comment:
You should implement an override to the `click_empty` proc because all this does is click if you're not next to a cade, or click on anything other than a cade.